src/features/quote/quote.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { MissingAuthorIdException } from 'src/common/filters/missingAuthorIdException.filter';
+import { Quote } from './quote.entity';
+import { Repository } from 'typeorm';
+import { InvalidAuthorIdException } from 'src/common/filters/invalidAuthorId.filter';
+
+@Injectable()
+export class QuoteService {
+  constructor(
+    @InjectRepository(Quote)
+    private readonly quoteRepository: Repository<Quote>,
+  ) {}
+
+  async getRandomQuoteFromAuthor(authorId: string | undefined) {
+    if (!authorId) {
+      throw new MissingAuthorIdException();
+    }
+
+    try {
+      const { quote, id: quoteId } = await this.quoteRepository
+        .createQueryBuilder('quote')
+        .where('quote.authorId = :authorId', { authorId })
+        .orderBy('RANDOM()')
+        .limit(1)
+        .getOne();
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({ authorId, quoteId, quote });
+        }, 5000);
+      });
+    } catch {
+      throw new InvalidAuthorIdException();
```

Comment:
This approach may not be ideal because other exceptions unrelated to an invalid author ID could occur, but it was the easiest way to handle this error

Reply:
that is fine